### PR TITLE
Clothes schema

### PR DESCRIPTION
**Pull request by rohanakut**
This PR is wrt to #580 . The PR creates a clothing schema which would be used by the preprocessor to generate data in a systematic format. The PR has been tested on online json schema validator and the format is as expected
---

## Required Information

- [x] I referenced the issue addressed in this PR.
- [x] I described the changes made and how these address the issue.
- [x] I described how I tested these changes.

## Coding/Commit Requirements

* [ ] I followed applicable coding standards (e.g., [PEP8](https://pep8.org/))
* [ ] I have not committed any models or other large files.

## New Component Checklist (**mandatory** for new microservices)

* [ ] I added an entry to `docker-compose.yml` and `build.yml`.
* [ ] I created A CI workflow under `.github/workflows`.

### CODE REVIEW

**Comment by JRegimbal** [comment]
Please link to the actual issue. "wrt" is not a linking keyword.

File: preprocessors/clothes.schema.json [hunk 1]
```diff
@@ -0,0 +1,44 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://image.a11y.mcgill.ca/preprocessors/clothes-detector.schema.json",
+    "type": "object",
+    "title": "Clothes Detector",
+    "description": "Preprocessor to detect the clothes worn by the individuals. The associated information can be found in preprocessor named 'ca.mcgill.a11y.image.preprocessor.clothesDetector'. ",
+    "properties":{
+        "clothes":{
+            "type":"array",
+            "description": "Contains a list of clothes worn by people",
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:10]: Please improve - is this just all the articles of clothing identified on all the people? This is ambiguous.

File: preprocessors/clothes.schema.json [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+            "items": {
+                "type": "object",
+                "properties": {
+                    "ID":{
+                        "description": "The ID of a person. The ID is generated by object detection preprocessor and is reused in this preprocessor",
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:15]: If this relies on person identification please specify this in the top-level description.

File: preprocessors/clothes.schema.json [hunk 3]
```diff
@@ -0,0 +1,44 @@
...
+                        "type": "integer"
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:16]: Please apply the same constraints as enforced in object detection for IDs.

File: preprocessors/clothes.schema.json [hunk 4]
```diff
@@ -0,0 +1,44 @@
...
+                    },
+                    "clothes":{
+                        "type":"array",
+                        "description": "List of outfits worn by the individual",
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:20]: What is an "outfit"?

File: preprocessors/clothes.schema.json [hunk 5]
```diff
@@ -0,0 +1,44 @@
...
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "outfit":{
+                                    "type":"string",
+                                    "description": "One of the outfit worn by the individual"
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:26]: This is a meaningless description. Please improve.

File: preprocessors/clothes.schema.json [hunk 6]
```diff
@@ -0,0 +1,44 @@
...
+                                },
+                                "confidence":{
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:28]: Improve description and actually add limits. I am skeptical that I will ever be seeing a confidence value of -26, but that's perfectly valid here.

File: preprocessors/clothes.schema.json [hunk 7]
```diff
@@ -0,0 +1,44 @@
+{
+    "$schema": "http://json-schema.org/draft-07/schema",
+    "$id": "https://image.a11y.mcgill.ca/preprocessors/clothes-detector.schema.json",
+    "type": "object",
+    "title": "Clothes Detector",
+    "description": "Preprocessor to detect the clothes worn by the individuals. The associated information can be found in preprocessor named 'ca.mcgill.a11y.image.preprocessor.clothesDetector'. ",
+    "properties":{
+        "clothes":{
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:8]: This is broken down by person, yet refers to the clothing worn by each person in the array. This doesn't make sense as a key name.

File: preprocessors/clothes.schema.json [hunk 8]
```diff
@@ -0,0 +1,44 @@
...
+                        }
+                    }
+                },
+                "required": [
+                    "ID",
+                    "clothes"
+                  ]
+            }
+        }
+    }
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:43]: Top level "clothes" key not required. Please fix.

**Review by JRegimbal — CHANGES_REQUESTED** [review]

File: preprocessors/clothes.schema.json [hunk 9]
```diff
@@ -0,0 +1,47 @@
...
+                        "type": "integer",
+                        "minimum": 0
+                    },
+                    "attire": {
+                        "type": "array",
+                        "description": "Lists the different clothes worn by an individual. Since one individual can wear multiple clothes such as a short sleeve top and an outerwear, we have used an array to describe the entire attire of the individual. Each element of an array would describe one piece of cloth worn by the individual.",
+                        "items": {
+                            "type": "object",
+                            "properties": {
+                                "clothes": {
```
Comment (JRegimbal) [inline preprocessors/clothes.schema.json:25]: Perhaps: article? type? What kind of values does this actually take? Is it something as simple as "t-shirt" or is this a full description like "a blue t-shirt with a green pocket"?

**Review by JRegimbal — CHANGES_REQUESTED** [review]

File: preprocessors/clothes.schema.json (hunk 9 above)
Comment (rohanakut) [reply preprocessors/clothes.schema.json:25]: so this would just generate t-shirt, dress, skirt.

File: preprocessors/clothes.schema.json (hunk 1 above)
Comment (JRegimbal) [reply preprocessors/clothes.schema.json:10]:
```suggestion
            "description": "Lists clothing identified for each person in the input.",
```

**Review by JRegimbal — CHANGES_REQUESTED** [review]

**Review by JRegimbal — APPROVED** [review]

**Pull request merged**